lib/internal/test_runner/test.js

```diff
@@ -998,7 +1000,21 @@ class Test extends AsyncResource {
       process.exit();
     }
   }
-
+  getFiles(files) {
```

Comment:
This does not seem like a good approach. This is only tracking `FileTest` instances. `FileTest`s are only used when `--test` is used, so I don't think this would work if you ran a single test file with coverage. `FileTest`s also shouldn't have subtests, so this seems like it might be doing extra work for nothing.

Reply:
Ahhh didn't knew that FileTests didn't have subtests. I've done this based on the properties that the Test class exposes. Anyway, based on your previous comment, I have to rethink this.